src/ring_impl.cpp

```diff
@@ -199,6 +218,23 @@ void BFring_impl::resize(BFsize contiguous_span,
 	_span       = new_span;
 	_stride     = new_stride;
 	_nringlet   = new_nringlet;
+	
+#if BF_NUMA_ENABLED
+	_size_log.update() << "space     : " << bfGetSpaceString(_space) << "\n"
```

Comment:
Can this and the similar code at line 109 be moved into a function to avoid duplication? The BF_NUMA_ENABLED condition could also be made tighter to further reduce duplication.